pkg/kubeapiserver/server/insecure_handler.go

```diff
@@ -20,14 +20,20 @@ import (
 	"net/http"
 
 	genericapifilters "k8s.io/apiserver/pkg/endpoints/filters"
+	"k8s.io/apiserver/pkg/features"
 	"k8s.io/apiserver/pkg/server"
 	genericfilters "k8s.io/apiserver/pkg/server/filters"
+	"k8s.io/apiserver/pkg/util/feature"
 )
 
 // BuildInsecureHandlerChain sets up the server to listen to http. Should be removed.
 func BuildInsecureHandlerChain(apiHandler http.Handler, c *server.Config) http.Handler {
 	handler := apiHandler
-	handler = genericfilters.WithMaxInFlightLimit(handler, c.MaxRequestsInFlight, c.MaxMutatingRequestsInFlight, c.LongRunningFunc)
+	if feature.DefaultFeatureGate.Enabled(features.APIPriorityAndFairness) {
+		handler = genericfilters.WithPriorityAndFairness(handler, c.LongRunningFunc, c.FlowControl)
```

Comment:
How will this work? We don't know the user on the insecure port.

Reply:
This conversation mostly happened at https://github.com/kubernetes/kubernetes/pull/85319#discussion_r366087149